### PR TITLE
feat(api-client): command palette navigation

### PR DESCRIPTION
**Pull request by antlio**
this pr adds the following enhancements:
- focus on first result when conducting command search
- return on global menu on delete key from a command
- enhances styles and fix border opacity

https://github.com/user-attachments/assets/3435ca91-ddd7-4fc8-83b9-061d4bb9e1c2

### CODE REVIEW

**Comment by hanspagel** [comment]
Great improvement! In Raycast, when I go back (backspace) it still has the search query and the previous results listed, I like that. :)

Screenshot 2024-10-01 at 14 38 21

**Review by hanspagel — APPROVED** [review]
Great improvement

File: packages/api-client/src/components/CommandPalette/TheCommandPalette.vue [hunk 1]
```diff
@@ -153,6 +161,21 @@ const openCommandPalette = ({
   nextTick(() => commandInputRef.value?.focus())
 }
 
+/** Focus on first result when conducting a search */
+watch(commandQuery, (newQuery) => {
+  if (newQuery && searchResultsWithPlaceholderResults.value.length > 0) {
+    selectedSearchResult.value = 0
+    nextTick(() => {
```
Comment (marclave) [inline packages/api-client/src/components/CommandPalette/TheCommandPalette.vue:168]: nit but we could do async await on next time to keep the indentation down :) but non blocking for this PR

```
watch(commandQuery, async (newQuery) => {
  if (newQuery && searchResultsWithPlaceholderResults.value.length > 0) {
    selectedSearchResult.value = 0
    await nextTick()

```

**Review by amritk — CHANGES_REQUESTED** [review]
Cannot delete any characters in the command pallets. We need to add a check for `string length === 0` before going back

**Comment by marclave** [comment]
feel free to merge after @amritk 's feedback is addressed @antlio <3

File: packages/api-client/src/components/CommandPalette/TheCommandPalette.vue [hunk 2]
```diff
@@ -153,6 +161,21 @@ const openCommandPalette = ({
...
+      commandRefs.value[selectedSearchResult.value]?.scrollIntoView({
+        behavior: 'smooth',
+        block: 'center',
```
Comment (hwkr) [inline packages/api-client/src/components/CommandPalette/TheCommandPalette.vue:171]: Should this be `nearest` like the search modal?

**Comment by antlio** [comment]
thanks for the great feedbacks, i added the following enhancements:
- maintain search query on back
- keep focus item on back
- only fire onDelete event on empty input

https://github.com/user-attachments/assets/348282d3-3c1e-4b54-a496-fffb1f98d09c

File: packages/api-client/src/components/CommandPalette/TheCommandPalette.vue (hunk 2 above)
Comment (amritk) [reply packages/api-client/src/components/CommandPalette/TheCommandPalette.vue]: Nearest all the things

**Review by amritk — APPROVED** [review]
Copilot says it's clear

**Pull request merged**